app/controllers/identity.js

```diff
@@ -14,16 +14,49 @@ export default class IdentityController extends Controller {
   @tracked isCopyClicked = false;
   @tracked isVerifyClicked = false;
   @tracked profileURL = this.model.profileURL || '';
+  @tracked saveDisabled = true;
+  @tracked generateChainCodeDisabled = this.model.profileURL === undefined;
+  @tracked checkboxDisabled =
+    this.generateChainCodeDisabled || this.model.chaincode === undefined;
+  @tracked linkDisabled = true;
 
   @action async handleRefresh() {
     window.location.reload();
   }
 
+  @action changeSaveDisabled() {
+    const isValidUrl = (urlString) => {
+      try {
+        return Boolean(new URL(urlString));
+      } catch (e) {
+        return false;
+      }
+    };
+    if (
+      this.profileURL === '' ||
+      this.profileURL === (this.model.profileURL || '') ||
+      !isValidUrl(this.profileURL)
+    ) {
+      this.saveDisabled = true;
+    } else {
+      this.saveDisabled = false;
+    }
+  }
+
+  @action changeLinkDisable() {
+    if (!this.isChecked) {
+      this.linkDisabled = false;
```

Comment:
Acc. to requirements link chain code should be enabled when -  the user has a profile URL and chain code in their user object and the checkbox is checked. That means all three conditions should be met. Are we checking all the three conditions here?, let me know if my understanding is wrong about the requirements

Reply:
Basically the checkbox will not be enabled if there is no profileURL or chaincode in the user object, so we don't need to check it here.